File: Semestrul 4/MPP/lab08/TriathlonNetworking/jsonprotocol/Response.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Triathlon.Model;

namespace Triathlon.Network.JsonProtocol
{
    [Serializable]
    public class Response
    {
        private ResponseType type;
        private string errorMessage;
        private RefereeDiscipline refereeDiscipline;
        private List<Result> results;
        private int totalPoints;

        public Response()
        {
        }

        public ResponseType Type
        {
            get { return type; }
            set { type = value; }
        }

        public RefereeDiscipline RefereeDiscipline
        {
            get { return refereeDiscipline; }
            set { refereeDiscipline = value; }
        }

        [JsonIgnore]
        public Event Event
        {
            get { return refereeDiscipline?.Event; }
        }

        [JsonIgnore]
        public Discipline Discipline
        {
            get { return refereeDiscipline.Discipline; }
        }

        public string ErrorMessage
        {
            get { return errorMessage; }
            set { errorMessage = value; }
        }

        [JsonIgnore]
        public Referee Referee
        {
            get { return refereeDiscipline?.Referee; }
        }

        public List<Result> Results
        {
            get { return results; }
            set { results = value; }
        }

        public int TotalPoints
        {
            get { return totalPoints; }
            set { totalPoints = value; }
        }

        public override string ToString()
        {
            return "Response{" +
                   "type=" + type +
                   ", errorMessage='" + errorMessage + '\'' +
                   ", referee=" + (refereeDiscipline != null ? Referee?.ToString() : "null") +
                   ", event=" + (refereeDiscipline != null ? Event?.ToString() : "null") +
                   ", discipline=" + (refereeDiscipline != null ? Discipline.ToString() : "null") +
                   ", results=" + results +
                   '}';
        }
    }
}

File: Semestrul 4/SDI/lab14/java/TriathlonNetworking/src/main/java/triathlon/network/jsonprotocol/Response.java
package triathlon.network.jsonprotocol;

import triathlon.model.*;

import java.io.Serializable;
import java.util.List;

public class Response implements Serializable {
    private ResponseType type;
    private String errorMessage;
    private RefereeDiscipline refereeDiscipline;
    private List<Result> results;
    private int totalPoints;

    public Response() {
    }

    public ResponseType getType() {
        return type;

    }

    public RefereeDiscipline getRefereeDiscipline() {
        return refereeDiscipline;
    }

    public void setRefereeDiscipline(RefereeDiscipline refereeDiscipline) {
        this.refereeDiscipline = refereeDiscipline;
    }

    public Event getEvent() {
        return refereeDiscipline.getEvent();
    }

    public Discipline getDiscipline() {
        return refereeDiscipline.getDiscipline();
    }

    public void setType(ResponseType type) {
        this.type = type;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public Referee getReferee() {
        return refereeDiscipline.getReferee();
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }
    public int getTotalPoints() {
        return totalPoints;
    }
    public void setTotalPoints(int totalPoints) {
        this.totalPoints = totalPoints;
    }

    @Override
    public String toString() {
        return "Response{" +
                "type=" + type +
                ", errorMessage='" + errorMessage + '\'' +
                ", referee=" + getReferee() +
                ", event=" + getEvent() +
                ", discipline=" + getDiscipline() +
                ", results=" + results +
                '}';
    }
}